CHANGELOG.md

```diff
@@ -4,6 +4,1647 @@ All notable changes to this project will be documented in this file.
+  >
+  > Related:https://github.com/rust-lang/rust/issues/137578
+
+### Styling
+
+- [2739391](https://github.com/ratatui/ratatui/commit/2739391950eddd129c25cd4ffddb28b99eed8cf5) *(uncategorized)* Use Module imports_granularity by @joshka in [#1728](https://github.com/ratatui/ratatui/pull/1728)
+
+  > I was swayed by the arguments about this made by the compiler team In
+  > <https://github.com/rust-lang/compiler-team/issues/750> and decided to
+  > look at how this organization affects ratatui. I found this reduces the
+  > number of lines across the codebase by about 350 and makes the imports
+  > more readable and definitely more greppable as you usually only have
+  > to read a single line. I've found in the past that maintaining imports
+  > regularly leads to merge conflicts which have to be resolved by hand
+  > and this change should reduce the likelihood of that happening.
+  >
+  > Main change is in rustfmt.toml, and the rest is just the result of
+  > running `cargo xtask format`.
+  >
+  > While implementing this, cargo machete brings up that the various
+  > backend crates are unused by the example crates.
+  >
+  > The re-export of each backend crate under ratatui is to make it possible
+  > for libs that rely on a specific version of ratatui to use the same
+  > version of the backend crate. Apps in general should use the backend
+  > crate directly rather than through ratatui as this is less confusing.
+  >
+  > - Removes all usages of `ratatui::{crossterm, termion, termwiz}`` in the
+  >   examples.
+  > - Adds the backend crate to the dependencies of the examples that use
+  >   the backend crate directly.
+
+### Miscellaneous Tasks
+
+- [1874b9d](https://github.com/ratatui/ratatui/commit/1874b9dd55284952931d58646904f0437e48ce76) *(uncategorized)* Move time to dev-dependencies by @j-g00da in [#1835](https://github.com/ratatui/ratatui/pull/1835)
+
+- [d88cd29](https://github.com/ratatui/ratatui/commit/d88cd2907906e38d73ac4359ece36527436d686a) *(uncategorized)* Add 'const' to functions where possible. by @MatrixFrog in [#1802](https://github.com/ratatui/ratatui/pull/1802)
+
+  > The Clippy check for this (missing_const_for_fn) is already enabled, but
+  > catches more cases in upcoming toolchain versions.
+  >
+  > This is part of the work to unblock #1727
+
+- [0f80c5e](https://github.com/ratatui/ratatui/commit/0f80c5e87eb4e47c818a6bd8b0002971920c65a0) *(uncategorized)* Use expect() instead of allow() for lint overrides by @cgzones in [#1786](https://github.com/ratatui/ratatui/pull/1786) [**breaking**]
+  >
+  > BREAKING CHANGE:MSRV is now 1.81
+
+### Security
+
+- [3745a67](https://github.com/ratatui/ratatui/commit/3745a67ba071d5a52e378af34f8409cd90912eb0) *(deps)* Bump rand from 0.9.0 to 0.9.1 by @dependabot[bot] in [#1804](https://github.com/ratatui/ratatui/pull/1804)
+
+  > Bumps [rand](https://github.com/rust-random/rand) from 0.9.0 to 0.9.1.
```

Comment:
Note: before we do a proper 0.30 release, we should fix up the template to properly kill the overly verbose dependabot spam.